### DISCUSSION TITLE
Add support for easily showing the tax for the default zone on the product page

### DISCUSSION THREAD

**Discussion by lukeholder** [craftcms/commerce#3324] [Ideas]
Move the tax calculation out of the default tax engine adjuster and into a service.

**Comment by JshGrn** [comment]
I haven't looked in detail on how Commerce handles its pricing, however I think that the tax calculation is on an entire cart. This makes it very tricky doing things which WooCommerce has no problem (!!!). I need to only and always show a VAT included price on a per product, but also need to know the total ex VAT and the total VAT for a summary prior to checkout.

This will be the usual checkout process for non b2b transactions within the UK and Europe so I am surprised nobody else has raised issues with it.

I have found numerous Stack Overflow posts which are simply multiplying the product price by the VAT rate, this is not good nor maintainable should the rate change in the backend.

I am having to do this on each time the price is required (line items in basket etc) and I hate it:

```
{% set productPrice = (item.purchasable.product.getTaxCategory().getTaxRates()[0].rate > 0 ? ((((item.purchasable.product.defaultVariant.price) / 100) * (item.purchasable.product.getTaxCategory().getTaxRates()[0].rate * 100)) + item.purchasable.product.defaultVariant.price) : item.purchasable.product.defaultVariant.price)|commerceCurrency %}
```

It's a mix and match from SO, other various posts I've found from Discord and other places on the internet coupled with some of my tweaks.

**Reply by JshGrn** [reply]
I also must add that our products only have one variant, and only serve one VAT zone (UK) so in our case its very simple and I am aware the above wont work outside of that scenario.
